app.js

```diff
@@ -1,17 +1,34 @@
 const express = require("express");
 const app = express();
-const { getTopics, getEndPoints } = require('./controllers/controller')
+const { getTopics, getEndPoints, getArticleID } = require('./controllers/controller')
 
 app.use(express.json());
 
 app.get('/api/topics', getTopics)
 
 app.get('/api', getEndPoints)
 
-app.all('/*', (req, res) => {res.status(404).send({msg: 'URL not found'})})
+app.get('/api/articles/:article_id', getArticleID)
+
+app.all('*', (req, res) => {res.status(404).send({msg: 'URL not found'})})
+
 
 app.use((err, req, res, next) => {
+    if (err.msg){
+        res.status(err.status).send({msg: err.msg})
+    } else {
+        next(err)
+    }
+})
 
+app.use((err, req, res, next) => {
+    if (err.code = '22P02'){
+        res.status(400).send({msg:'INVALID INPUT'})
+    } else {
+        next(err)
+    }
```

Comment:
Good error handlers set up, you should use === instead here though. It's working at the moment because you only have one instance of an error code, you are reassigning the value here even if it is that code beforehand.